compaction/compaction_manager.cc

```diff
@@ -1379,13 +1379,23 @@ class offstrategy_compaction_task_executor : public compaction_task_executor, pu
                 }));
         };
 
-        auto get_next_job = [&] () -> std::optional<sstables::compaction_descriptor> {
-            auto desc = t.get_compaction_strategy().get_reshaping_job(get_reshape_candidates(), t.schema(), sstables::reshape_mode::strict);
-            return desc.sstables.size() ? std::make_optional(std::move(desc)) : std::nullopt;
+        auto get_next_job = [&] () -> future<std::optional<sstables::compaction_descriptor>> {
+            auto candidates = get_reshape_candidates();
+            if (candidates.empty()) {
+                co_return std::nullopt;
+            }
+            // all sstables added to maintenance set share the same underlying storage.
+            auto& storage = candidates.front()->get_storage();
+            sstables::reshape_config cfg = {
+                .mode = sstables::reshape_mode::strict,
+                .free_storage_space = co_await storage.free_space(),
```

Comment:
We should divide by the number of shards. Otherwise, each shard will consume its 10% , and all of them together will consume 880%.

Reply:
twcs reshape is adjusting it, but we can do it in free_space() too. it's not perfect that we're averaging the remaining space to shards, as it assumes shards are uniform in space utilization. not that it's very important here, but I was thinking about other cases we might want to inspect (one day) space utilization. and possibly we don't want to impose a constraint that space is equally divided among shards. IIRC, I think we're facing a issue of this nature in I/O sched.

Reply:
I don't understand whether the code is correct or not and what's the relation to the scheduler (that manages bandwidth, not space).

Reply:
this is already changed in next version. please forget about the comparison, it was based on my weak understanding.